src/lucky/base_http_client.cr

```diff
@@ -1,28 +1,104 @@
 require "http/client"
 
-class Lucky::BaseHTTPClient
+# A client for making HTTP requests
+#
+# Makes it easy to pass params, use Lucky route helpers, and chain header methods.
+abstract class Lucky::BaseHTTPClient
+  private getter client
+
   @client : HTTP::Client
 
   def initialize(host = Lucky::Server.settings.host, port = Lucky::Server.settings.port)
     @client = HTTP::Client.new(host, port: port)
   end
 
-  def get(path : String, headers : HTTP::Headers? = nil, params : HTTP::Params? = nil)
-    if params
-      path = path + "?#{params}"
+  {% for method in [:get, :put, :patch, :post] %}
+    def self.{{ method.id }}(*args, **named_args)
+      new.{{ method.id }}(*args, **named_args)
     end
-    @client.get(path, headers: headers)
-  end
+  {% end %}
 
-  {% for method in [:put, :patch, :post] %}
+  # Set headers for requests
+  #
+  # ```
+  # # `content_type` will be normalized to `content-type`
+  # AppClient.new.headers(content_type: "application/json")
+  #
+  # # You can also use string keys if you want
+  # AppClient.new.headers("Content-Type": "application/json")
+  # ```
 
-    def {{method.id}}(path : String, body : Hash(String, String), headers : HTTP::Headers? = nil)
-      @client.{{method.id}}(path, headers: headers, form: body)
+  # The header call is chainable and returns the client:
+  #
+  # ```
+  # # content_type will be normalized to `content-type`
+  # AppClient.new
+  #   .headers(content_type: "application/json")
+  #   .headers(accept: "text/plain")
+  #   .get("/some-path")
+  # ```
+  #
+  # You can also set up headers in `initialize` or in instance methods:
+  #
+  # ```
+  # class AppClient < Lucky::BaseHTTPClient
+  #   def initialize
+  #     headers(content_type: "application/json")
+  #   end
+  #
+  #   def accept_plain_text
+  #     headers(accept: "text/plain")
+  #   end
+  # end
+  #
+  # AppClient.new
+  #   .accept_plain_text
+  #   .get("/some-path")
+  # ```
+  def headers(**header_values)
+    @client.before_request do |request|
+      header_values = header_values
+        .to_h
+        .transform_keys(&.gsub("_", "_"))
```

Comment:
Did you mean

```suggestion
        .transform_keys(&.gsub("_", "-"))
```

Reply:
Yeah :D And I had to tweak this a bit because there was a compile error